spec/views/events/index.html.erb_spec.rb

```diff
@@ -33,4 +33,25 @@
     expect(rendered).to have_link(href: edit_event_path(@event))
     expect(rendered).to have_link(href: event_path(@event), class: 'btn-danger')
   end
+
+  it "should not display apply button if application deadline is over" do
+    @event.application_deadline = Date.yesterday
+    assign(:events, [@event])
+    [:pupil, :coach, :organizer].each do |role|
```

Comment:
I would still leave it like it is, since somewhere in the future it might be necessary to let roles different from pupil apply to events too. I guess removing roles because they shouldn't be able to apply would mix logic for two things in a bad way.